client/static/js/index.js

```diff
@@ -0,0 +1,83 @@
+const searchResults = document.getElementById('results');
+const logo = document.getElementById('logo');
+
+//function to show 10 search results
+function getSearchResult(e) {
+    //handling spaces in searches
+    const searchString = searchBar.value.replace(/\s/g, "_");
+
+    //hiding logo and showing results
+    searchEngine.className = "moveUp";
+    logo.className = "hide";
+    searchResults.className = "";
+
+    fetch(`http://localhost:3000/${searchString}`)
+       .then(resp => resp.json())
+       .then(resp => { 
+            if(resp.error) {
+                //displays message if no results are found
+                const noResults = document.createElement("H1");
+                const noResultsMessage = document.createTextNode(`No results found for \"${searchString}\".`);
+                noResults.appendChild(noResultsMessage);
+                searchResults.appendChild(noResults);
+            }
+
+            for(let i = 0; i < resp.length; i++) //loop through all array elements
+            {
+                //adding website name to search results
+                const websiteName = document.createElement("A");
+                const websiteNameText = document.createTextNode(`${resp[i].header}`);
+                websiteName.setAttribute("href", `${resp[i].link}`);
+                websiteName.appendChild(websiteNameText);
+                searchResults.appendChild(websiteName);
+                
+                //adding website info to search results
+                const websiteInfo = document.createElement("P");
+                const websiteInfoText = document.createTextNode(`${resp[i].bodyText}`);
+                websiteInfo.appendChild(websiteInfoText);
+                searchResults.appendChild(websiteInfo);
+            };
+        })
+        .catch(err => { 
+            console.log(err) });
+}
+
+//function to show a random result
+function getSurpriseResult(e) {
+    const searchString = searchBar.value;
+
+    searchEngine.className = "hide";
+    searchResults.className = "";
+
+    fetch(`http://localhost:3000/${searchString}/random`)
+       .then(resp => resp.json())
+       .then(data => {
+        if(data.error) {
+            //displays message if no results are found
+            const noResults = document.createElement("H1");
+            const noResultsMessage = document.createTextNode(`No results found for \"${searchString}\".`);
+            noResults.appendChild(noResultsMessage);
+            searchResults.appendChild(noResults);
+        }
+        else {
+            //takes you to a random page based on the search term
+            window.location.replace(`${data.link}`);
+        }
+       })
+       .catch(err => { 
+        console.log(err) });
+}
+
+searchButton.addEventListener('click', e => {
+    e.preventDefault();
+    getSearchResult(e);
+});
+
+surpriseButton.addEventListener('click', e => {
+    e.preventDefault();
+    getSurpriseResult(e);
+});
```

Comment:
Was redirected to a random site, but it wouldn't let me go back to the search bar - it could be a nice touch to allow the user to be able to move back instead of having to restart the client

This is not the case when I click a website from the list of links

Reply:
Thanks for pointing this out - I've fixed this now.